pkg/controllers/termination/terminator/terminator.go

```diff
@@ -59,60 +61,53 @@ func (t *Terminator) Cordon(ctx context.Context, node *v1.Node) error {
 		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(stored)); err != nil {
 			return err
 		}
-		logging.FromContext(ctx).Infof("cordoned node")
+		logging.FromContext(ctx).Infof("cordoned")
 	}
 	return nil
 }
 
 // Drain evicts pods from the node and returns true when all pods are evicted
 // https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
 func (t *Terminator) Drain(ctx context.Context, node *v1.Node) error {
-	// Get evictable pods
-	pods, err := t.getPods(ctx, node)
-	if err != nil {
-		return fmt.Errorf("listing pods for node, %w", err)
-	}
-	var podsToEvict []*v1.Pod
-	// Skip node due to pods that are not able to be evicted
-	for _, p := range pods {
-		// Ignore if unschedulable is tolerated, since they will reschedule
-		if podutil.ToleratesUnschedulableTaint(p) {
-			continue
-		}
-		// Ignore static mirror pods
-		if podutil.IsOwnedByNode(p) {
-			continue
-		}
-		podsToEvict = append(podsToEvict, p)
+	podList := &v1.PodList{}
+	if err := t.kubeClient.List(ctx, podList, client.MatchingFields{"spec.nodeName": node.Name}); err != nil {
+		return fmt.Errorf("listing pods on node, %w", err)
 	}
-	// Enqueue for eviction
-	t.evict(podsToEvict)
-
-	if len(podsToEvict) > 0 {
-		return NewNodeDrainError(fmt.Errorf("%d pods are waiting to be evicted", len(podsToEvict)))
+	pods := lo.Reject(lo.ToSlicePtr(podList.Items), func(p *v1.Pod, _ int) bool { return podutil.IsNotEvictable(p) }) // Ignore pods that can't be drained
+	if len(pods) == 0 {
+		return nil
 	}
-	return nil
+	// Evict, respecting PDBs
```

Comment:
what if PDB has broken pods, are we going to force delete those broken pods somewhere?